### PR TITLE
scheduler: handle and log preemption return error

### PR DESCRIPTION
**Pull request by wgliang**
**What type of PR is this?**
/kind cleanup
/sig scheduling

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -466,7 +466,9 @@ func (sched *Scheduler) scheduleOne() {
 					" No preemption is performed.")
 			} else {
 				preemptionStartTime := time.Now()
-				sched.preempt(pod, fitError)
+				if err := sched.preempt(pod, fitError); err != nil {
+					klog.Warningf("%v/%v preempting lower priority pods fail,%v", pod.Namespace, pod.Name, err)
```
Comment (neolit123) [inline pkg/scheduler/scheduler.go]: please change to `... pods fail: %v`

Comment (neolit123) [reply pkg/scheduler/scheduler.go]: what would be the report rate of this warning?
does it make sense to only show it in verbose mode - e.g. `V(2)`?

Comment (ravisantoshgudimetla) [reply pkg/scheduler/scheduler.go]: Agreed, increasing log level will ensure we are not aggressively filling logs.

Comment (ravisantoshgudimetla) [inline pkg/scheduler/scheduler.go]: Shouldn't we increment scheduling failures here too? line https://github.com/kubernetes/kubernetes/pull/75369/files#diff-0260dd7e154d411be493ada0753dc547L479

Comment (wgliang) [reply pkg/scheduler/scheduler.go]: Yes, it will always run metrics.PodScheduleFailures.Inc().

File: pkg/scheduler/scheduler.go [hunk 2]
```diff
@@ -287,17 +287,17 @@ func (sched *Scheduler) schedule(pod *v1.Pod) (core.ScheduleResult, error) {
 // preempt tries to create room for a pod that has failed to schedule, by preempting lower priority pods if possible.
 // If it succeeds, it adds the name of the node where preemption has happened to the pod annotations.
 // It returns the node name and an error if any.
```
Comment (hex108) [inline pkg/scheduler/scheduler.go]: This line also needs change: It returns an error if any.

File: pkg/scheduler/scheduler.go [hunk 3]
```diff
@@ -466,7 +466,9 @@ func (sched *Scheduler) scheduleOne() {
 					" No preemption is performed.")
 			} else {
 				preemptionStartTime := time.Now()
-				sched.preempt(pod, fitError)
+				if err := sched.preempt(pod, fitError); err != nil {
```
Comment (hex108) [inline pkg/scheduler/scheduler.go]: How about keeping return (node, err) same as before, and printing it if err == nil? It might be useful for debugging.

**Review by Huang-Wei — COMMENTED** [review]
LGTM.

It seems the only caller of `preempt()` is `scheduleOne()` and it doesn't rely on the return values.

Will leave /approval to @bsalamat.

File: pkg/scheduler/scheduler.go (hunk 2 above)
Comment (Huang-Wei) [reply pkg/scheduler/scheduler.go]: +1. Please address.

Comment (wgliang) [reply pkg/scheduler/scheduler.go]: DONE

File: pkg/scheduler/scheduler.go [hunk 4]
```diff
@@ -466,7 +466,9 @@ func (sched *Scheduler) scheduleOne() {
...
+					klog.Warningf("%v/%v preempting lower priority pods fail: %v", pod.Namespace, pod.Name, err)
```
Comment (bsalamat) [inline pkg/scheduler/scheduler.go]: The `preempt` function already logs error messages. We don't need to double log here.

**Review by bsalamat — COMMENTED** [review]
Thanks, @wgliang. Looks fine to me. A minor comment and please rebase.

**Comment by wgliang** [comment]
DONE, Thanks for your review. PTAL @Huang-Wei @bsalamat

**Comment by mrkm4ntr** [comment]
@wgliang How about using return value to determine label of this metrics?
https://github.com/kubernetes/kubernetes/blob/7dc5d1f0646e0db0c077db578cf4b0855d8c799a/pkg/scheduler/metrics/metrics.go#L188-L193
If `PreemptionAttempts` has labels like `scheduleAttempts`, it might be helpful.
https://github.com/kubernetes/kubernetes/blob/7dc5d1f0646e0db0c077db578cf4b0855d8c799a/pkg/scheduler/metrics/metrics.go#L52-L63

**Pull request closed without merging**